Source/Plugin.BLE.iOS/Adapter.cs

```diff
@@ -366,5 +365,141 @@ public static List<AdvertisementRecord> ParseAdvertismentData(NSDictionary adver
 
             return records;
         }
+
+        /// <summary>
+        /// See: https://developer.apple.com/library/archive/documentation/NetworkingInternetWeb/Conceptual/CoreBluetooth_concepts/Art/ReconnectingToAPeripheral_2x.png for a chart of the flow.
+        /// </summary>
+        protected override async Task<IDevice> ConnectNativeAsync(string friendlyName, Guid uuid, Func<IDevice, bool> deviceFilter, CancellationToken cancellationToken = default)
+        {
+            if (uuid == Guid.Empty)
+            {
+                // If we do not have an uuid scan and connect
+                return await ScanAndConnect(friendlyName, deviceFilter);
+            }
+            else
+            {
+                //FYI attempted to use tobyte array insetead of string but there was a problem with byte ordering Guid->NSUuid
+                var nsuuid = new NSUuid(uuid.ToString());
+
+                // If we have an uuid, check if the system can find the device.
+                var peripheral = TryToRetrieveKnownPeripheral(nsuuid);
+                if (peripheral == null)
+                {
+                    // The device haven't been found. We'll try to scan and connect.
+                    return await ScanAndConnect(friendlyName, deviceFilter);
```

Comment:
Async postfix?

Reply:
We have no async police here in the plugin 😭. Good thing that we have your sharp eyes instead! It is fixed